GEOSwift/GEOS.swift

```diff
@@ -184,9 +184,10 @@ public struct CoordinatesCollection: SequenceType {
     
     public func generate() -> AnyGenerator<Coordinate> {
         var index: UInt32 = 0
-        return anyGenerator {
+        return AnyGenerator {
+            index += 1
```

Comment:
This is wrong: `index` should be incremented AFTER accessing the element, not BEFORE, something like this:

```swift
    public func generate() -> AnyGenerator<T> {
        var index: Int32 = 0
        return AnyGenerator {
            if index < self.count {
                let item = self[index]
                index = index + 1
                return item
            }
            return nil
        }
    }
```